config.c

```diff
@@ -1911,19 +1911,78 @@ static int readConfigFile(const char *configFile, struct logInfo *defConfig)
                             const struct logInfo *log;
                             struct stat sb_glob;
                             int add_file = 1;
+                            char *copy, *baseName, *dirName, *resolvedDir, *resolved_path;
+                            unsigned k;
 
                             /* if we glob directories we can get false matches */
                             if (!lstat(globResult.gl_pathv[glob_count], &sb_glob) &&
                                     S_ISDIR(sb_glob.st_mode)) {
                                 continue;
                             }
 
+                            /* Resolves the directory but not the filename, to be able to detect symlinks */
+                            copy = strdup(globResult.gl_pathv[glob_count]);
+                            if (copy == NULL) {
+                                message_OOM();
+                                logerror = 1;
+                                goto duperror;
+                            }
+                            /* Because dirname() modifies the path, order is important */
+                            baseName = basename(copy);
+                            dirName = dirname(copy);
```

Comment:
Both `basename()` and `dirname()` might modify their argument, so I don't think this is safe.